frame/elections-phragmen/src/lib.rs

```diff
@@ -284,6 +291,43 @@ pub mod pallet {
 				Weight::zero()
 			}
 		}
+
+		fn integrity_test() {
+			let block_weight = T::BlockWeights::get().max_block;
+			// mind the order.
+			let election_weight = T::WeightInfo::election_phragmen(
+				T::MaxCandidates::get(),
+				T::MaxVoters::get(),
+				T::MaxVotesPerVoter::get() * T::MaxVoters::get(),
+			);
+
+			let to_seconds = |w: &Weight| {
+				w.ref_time() as f32 /
+					frame_support::weights::constants::WEIGHT_REF_TIME_PER_SECOND as f32
+			};
+
+			frame_support::log::debug!(
+				target: LOG_TARGET,
+				"election weight {}s ({:?}) // chain's block weight {}s ({:?})",
+				to_seconds(&election_weight),
+				election_weight,
+				to_seconds(&block_weight),
+				block_weight,
+			);
+			if election_weight.any_gt(block_weight) {
+				frame_support::log::error!(
```

Comment:
Should this not bee an `assert!`? Otherwise the test will not fail.

Reply:
Thing is, for a solo-chain, this is more or less acceptable to have an overweight election. For a parachain, it is not. 

Ideally, for this we'd have @sam0x17's 

```
if parachain_build {} else {} 
```

For now I will move it to assert, better safe than sorry.